centreon/www/class/centreonLog.class.php

```diff
@@ -129,77 +147,379 @@ public static function singleton($uid = 0)
     }
 }
 
+/**
+ * Class
+ *
+ * @class CentreonLog
+ */
 class CentreonLog
 {
-    private $errorType;
-    private $path;
+    // Level Types from \Psr\Log\LogLevel
+    public const LEVEL_DEBUG = LogLevel::DEBUG;
+    public const LEVEL_NOTICE = LogLevel::NOTICE;
+    public const LEVEL_INFO = LogLevel::INFO;
+    public const LEVEL_WARNING = LogLevel::WARNING;
+    public const LEVEL_ERROR = LogLevel::ERROR;
+    public const LEVEL_CRITICAL = LogLevel::CRITICAL;
+    public const LEVEL_ALERT = LogLevel::ALERT;
+    public const LEVEL_EMERGENCY = LogLevel::EMERGENCY;
 
-    /*
-     * Constructor
+    /**
+     * Log files
      */
+    public const TYPE_LOGIN = 1;
+    public const TYPE_SQL = 2;
+    public const TYPE_LDAP = 3;
+    public const TYPE_UPGRADE = 4;
+    public const TYPE_PLUGIN_PACK_MANAGER = 5;
 
-    public function __construct($customLogs = array())
-    {
-        $this->errorType = array();
+    private const DEFAULT_LOG_FILES = [
+        self::TYPE_LOGIN => 'login.log',
+        self::TYPE_SQL => 'sql-error.log',
+        self::TYPE_LDAP => 'ldap.log',
+        self::TYPE_UPGRADE => 'upgrade.log',
+        self::TYPE_PLUGIN_PACK_MANAGER => 'plugin-pack-manager.log',
+    ];
 
-        /*
-         * Init log Directory
-         */
-        $this->path = _CENTREON_LOG_;
+    /** @var array<int,string> */
+    private array $logFileHandler;
 
-        $this->errorType[1] = $this->path . "/login.log";
-        $this->errorType[2] = $this->path . "/sql-error.log";
-        $this->errorType[3] = $this->path . "/ldap.log";
-        $this->errorType[4] = $this->path . "/upgrade.log";
-        $this->errorType[5] = $this->path . '/plugin-pack-manager.log';
+    /** @var string */
+    private string $pathLogFile;
 
-        foreach ($customLogs as $key => $value) {
-            if (!preg_match('@' . $this->path . '@', $value)) {
-                $value = $this->path . '/' . $value;
-            }
-            $this->errorType[$key] = $value;
+    /**
+     * CentreonLog constructor
+     *
+     * @param array $customLogFiles
+     * @param string $pathLogFile
+     */
+    public function __construct(array $customLogFiles = [], string $pathLogFile = '')
+    {
+        $this->setPathLogFile(empty($pathLogFile) ? _CENTREON_LOG_ : $pathLogFile);
+        // push default logs in log file handler
+        foreach (self::DEFAULT_LOG_FILES as $idLogType => $logFileName) {
+            $this->pushLogFileHandler($idLogType, $logFileName);
+        }
+        // push custom logs in log file handler
+        foreach ($customLogFiles as $idLogType => $logFileName) {
+            $this->pushLogFileHandler($idLogType, $logFileName);
```

Comment:
```suggestion
        foreach (self::DEFAULT_LOG_FILES as $idLogType => $logFileName) {
            $this->pushLogFileHandler($idLogType, $logFileName);
        }
        // push custom logs in log file handler
        foreach ($customLogFiles as $logTypeId => $logFileName) {
            $this->pushLogFileHandler($lodTypeId, $logFileName);
```
Just a suggest to be more convenient with our variable name convetion

Reply:
fixed :)